src/userSync.js

```diff
@@ -0,0 +1,192 @@
+// Let setQueue() set the defaults
+let queue;
+setQueue();
+
+// Since user syncs require cookie access we want to prevent sending syncs if cookies are not supported
+let cookiesAreSupported = !utils.isSafariBrowser() && utils.cookiesAreEnabled();
+// Whether or not user syncs have been trigger on this page load
+let hasFired = false;
+// How many bids for each adapter
+let numAdapterBids = {};
+
+// Merge the defaults with the user-defined config
+let userSyncConfig = Object.assign($$PREBID_GLOBAL$$.userSync || {},
+  userSyncDefaultConfig);
+
+/**
+ * @function setQueue
+ * @summary Sets the default empty queue
+ * @private
+ */
+function setQueue() {
+  queue = {
+    image: []
+  }
+}
+
+/**
+ * @function fireSyncs
+ * @summary Trigger all user syncs in the queue
+ * @private
+ */
+function fireSyncs() {
+  if (!userSyncConfig.syncEnabled || !cookiesAreSupported || hasFired) {
+    return;
+  }
+
+  try {
+    if (!userSyncConfig.pixelEnabled) {
+      return;
+    }
+    // Randomize the order of the pixels before firing
+    // This is to avoid giving any bidder who has registered multiple syncs
+    // any preferential treatment and balancing them out
+    utils.shuffle(queue.image).forEach((sync) => {
+      let bidderName = sync[0];
+      let trackingPixelUrl = sync[1];
+      utils.logMessage(`Invoking image pixel user sync for bidder: ${bidderName}`);
+      // Create image object and add the src url
+      userSync.createImgObject(trackingPixelUrl);
+    });
+    // Reset the user sync queue
+    userSync.resetQueue();
+    hasFired = true;
+  }
+  catch (e) {
+    utils.logError('Error firing user syncs', e);
+  }
+}
+
+/**
+ * @function hideAndIdElem
+ * @summary Modifies a DOM element to be hidden from user sight and adds a unique ID
+ * @private
+ * @params {object} elementNode A valid DOM element
+ * @returns {object} A valid DOM element
+ */
+function hideAndIdElem(elementNode) {
+  elementNode = elementNode.cloneNode();
+  elementNode.height = 0;
+  elementNode.width = 0;
+  elementNode.style.display = 'none';
+  elementNode.id = utils.getUniqueIdentifierStr();
+  return elementNode;
+}
+
+/**
+ * @function incrementAdapterBids
+ * @summary Increment the count of user syncs queue for the adapter
+ * @private
+ * @params {object} numAdapterBids The object contain counts for all adapters
+ * @params {string} bidder The name of the bidder adding a sync
+ * @returns {object} The updated version of numAdapterBids
+ */
+function incrementAdapterBids(numAdapterBids, bidder) {
+  if (!numAdapterBids[bidder]) {
+    numAdapterBids[bidder] = 1;
+  } else {
+    numAdapterBids[bidder] += 1;
+  }
+  return numAdapterBids;
+}
+
+/**
+ * @function createImgObject
+ * @summary Create an img DOM element for sending a pixel. Made public for test purposes
+ * @private
+ * @params {string} url The URL for the image pixel
+ * @returns {object} A valid DOM element
+ */
+userSync.createImgObject = (url) => {
```

Comment:
There are actually a couple things we could do here.

First of all, I think there are a number of functions in utils that are used solely for user sync that can likely go away. I will double check when I get back in the office on Thursday and remove any unnecessary functions. I can replace the utils version with this. We also no longer need to remove it onload since we are not actually adding the elem to the DOM any more.

Second, we can leave it in the userSync module since it is intended for use with user sync and likely will serve no other module. We would still remove any unnecessary functions from utils.

Thoughts?